native/src/main/java/io/ballerina/lib/data/jsondata/json/Native.java

```diff
@@ -96,6 +97,14 @@ public static Object toJson(Object value) {
     }
 
     public static Object toJson(Object value, Set<Object> visitedValues) {
+        if (value == null) {
+            return null;
+        }
+
+        if (TypeUtils.getType(value).getTag() < 7) {
```

Comment:
Shall we add switch tag or a function like `getSimpleType` in here.
Otherwise this is not easy to understand